src/html.cr

```diff
@@ -34,15 +34,260 @@ module HTML
   def self.unescape(string : String)
     return string unless string.includes? '&'
 
-    string.gsub(/&(apos|amp|quot|gt|lt|nbsp|\#[0-9]+|\#[xX][0-9A-Fa-f]+);/) do |string, _match|
+    string.gsub(/&(\w{2,8}|\#[0-9]+|\#[xX][0-9A-Fa-f]+);/) do |string, _match|
       match = _match[1]
       case match
-      when "apos" then "'"
-      when "amp"  then "&"
-      when "quot" then "\""
-      when "gt"   then ">"
-      when "lt"   then "<"
-      when "nbsp" then " "
+      when "apos"     then "'"
+      when "nbsp"     then '\u{A0}'
+      when "iexcl"    then '\u{A1}'
```

Comment:
In any case I'd put this hash in another file, load it at runtime once and do a hash lookup to simplify the code.